docs/source/executing-operations/query-batching.mdx

```diff
@@ -0,0 +1,208 @@
+description: Receive query batches with the Apollo Router  
+---
+
+> **Experimental Feature**  
+> The Apollo Router provides *experimental* support for client query batching.
+
+Learn about query batching and how to configure the Apollo Router to receive query batches.
+
+## About query batching
+
+Modern applications use several operations to render a single page. This is usually the result of a component-based architecture where individual micro-frontends (MFE) make requests separately to fetch data relevant to them. Not only does this cause a performance overhead (i.e. different components may be requesting the same data), it can also cause a consistency issue. To combat this, MFE-based UIs batch multiple client queries and operations, issued close together, into one HTTP request. This is supported in Apollo Client and Apollo Server.
+
+The Apollo Router now supports client request batching. If you’re using Apollo Client, you can leverage the built-in support for batching to reduce the number of individual requests sent to the router.
+
+Once configured, Apollo Client will automatically combine multiple operations into a single HTTP request. The number of operations within a batch is client configurable, including the maximum number of operations in a batch and the maximum duration to wait for operations to accumulate before sending the batch request. 
+
+The Apollo Router must be configured to receive batch requests, otherwise it rejects them. When processing a batch request, the router deserializes and processes each operation of a batch independently, and it responds to the client only after all operations of the batch have been completed. Each operation executes concurrently with respect to other operations in the batch.
+
+## Configure query batching
+
+Both a client and the Apollo Router need to be configured to support query batching.
+
+### Configure router
+
+By default, receiving client request batches is disabled in Apollo Router. 
+
+To enable query batching, set the following fields in your `router.yaml` configuration file:
+
+```yaml title="router.yaml"
+experimental_batching:
+  enabled: true
+  mode: batch_http_link
+```
+
+| Attribute | Description | Valid Values | Default Value |
+| :-- | :-- | :-- | :-- |
+| `enabled` | Flag to enable reception of client batches | boolean | `false` |
+| `mode` | Supported client batching mode | `batch_http_link`:  the client uses Apollo Link and its [`BatchHttpLink`](/react/api/link/apollo-link-batch-http) link. | No Default |
+
+### Configure client
+
+To enable batching in a client, configure `BatchHttpLink`. For details on implementing `BatchHttpLink`, see [batching operations](https://www.apollographql.com/docs/react/api/link/apollo-link-batch-http/).
+
+### Configuration compatibility
+
+If the router receives a query batch from a client, and batching is *not* enabled, the router sends a `BATCHING_NOT_ENABLED` error to the client, which clearly identifies why the query batch failed.
```

Comment:
The error does contain additional information. It would look something like this:
```
{
  "errors": [
    {
      "message": "Invalid GraphQL request",
      "extensions": {
        "details": "batching not enabled",
        "code": "BATCHING_NOT_ENABLED"
      }
    }
  ]
}
```
However, I agree that the dependent clause is superfluous.